src/k8s-extension/azext_k8s_extension/partner_extensions/Dapr.py

```diff
@@ -4,43 +4,143 @@
 # --------------------------------------------------------------------------------------------
 
 # pylint: disable=unused-argument
-# pylint: disable=line-too-long
 # pylint: disable=too-many-locals
+# pylint: disable=too-many-instance-attributes
 
+from typing import Tuple
+
+from azure.cli.core.azclierror import InvalidArgumentValueError
+from knack.log import get_logger
+from knack.prompting import prompt, prompt_y_n
+
+from ..vendored_sdks.models import Extension, Scope, ScopeCluster
 from .DefaultExtension import DefaultExtension
 
-from ..vendored_sdks.models import (
-    Extension,
-)
+logger = get_logger(__name__)
 
 
 class Dapr(DefaultExtension):
     def __init__(self):
+        self.TSG_LINK = "https://docs.microsoft.com/en-us/azure/aks/dapr"
+        self.DEFAULT_RELEASE_NAME = 'dapr'
+        self.DEFAULT_RELEASE_NAMESPACE = 'dapr-system'
+        self.DEFAULT_RELEASE_TRAIN = 'stable'
+        self.DEFAULT_CLUSTER_TYPE = 'managedclusters'
+        self.DEFAULT_HA = 'true'
+
         # constants for configuration settings.
-        self.CLUSTER_TYPE = 'global.clusterType'
+        self.CLUSTER_TYPE_KEY = 'global.clusterType'
+        self.HA_KEY_ENABLED_KEY = 'global.ha.enabled'
+        self.SKIP_EXISTING_DAPR_CHECK_KEY = 'skipExistingDaprCheck'
+        self.EXISTING_DAPR_RELEASE_NAME_KEY = 'existingDaprReleaseName'
+        self.EXISTING_DAPR_RELEASE_NAMESPACE_KEY = 'existingDaprReleaseNamespace'
+
+        # constants for message prompts.
+        self.MSG_IS_DAPR_INSTALLED = "Is Dapr already installed in the cluster?"
+        self.MSG_ENTER_RELEASE_NAME = "Enter the Helm release name for Dapr, "\
+            f"or press Enter to use the default name [{self.DEFAULT_RELEASE_NAME}]: "
+        self.MSG_ENTER_RELEASE_NAMESPACE = "Enter the namespace where Dapr is installed, "\
+            f"or press Enter to use the default namespace [{self.DEFAULT_RELEASE_NAMESPACE}]: "
+        self.MSG_WARN_EXISTING_INSTALLATION = "The extension will be installed on your existing Dapr installation. "\
+            f"Note, if you have updated the default values for global.ha.* or dapr_placement.* in your existing "\
+            "Dapr installation, you must provide them in the configuration settings. Failing to do so will result in"\
```

Comment:
By 'configuration settings', if you are referring to the parameter to the cli command, it would be good to use the same name of this parameter '--configuration-settings'

Reply:
Fixed!